AppIntegrity/Sources/AppIntegrity/URLRequest+Integrity.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+extension URLRequest {
+    static func clientAttestation(token: String) throws -> URLRequest {
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = "app-integrity-spike.mobile.dev.account.gov.uk"
+        components.path = "/client-attestation"
+        components.queryItems = [URLQueryItem(name: "device", value: "ios")]
+
+        let encoder = JSONEncoder()
+
+        var request = URLRequest(url: components.url!)
+        request.httpMethod = "POST"
+        request.httpBody = try encoder.encode(ClientAssertionRequest(jwk: token))
+        request.addValue(token, forHTTPHeaderField: "X-Firebase-AppCheck")
```

Comment:
I'm a little confused as the TD says:
"App calls Mobile Backend /client-attestation endpoint with Firebase App Check token and public app instance key. Mobile Backend validates App Check token claims and signature to validate that Firebase App Check has signed the token and that it is for the One Login app."
We seem to be passing the App Check token twice, once in the header for "X-Firebase-AppCheck" and again in the body for `{"jwk": "token"}`.
I can see on the TD it mentions validating the JWT claims which pertain to Firebase:
![image 2024-10-01 at 11 48 48](https://github.com/user-attachments/assets/758873cc-ed76-4675-b278-60071ca2d89a)
so not sure where we are including the public key from the app-generated key pair?

Reply:
Yes, the body is currently wrong, this is WIP 😅 
See here: https://govukverify.atlassian.net/browse/DCMAW-10320

Reply:
I think PoP JWT should be sent to the `/token` endpoint, I thought this was supposed the public key we generate in-app.
Anyway, something we can hash out later 👍